policies/ControllerReadOnlyFileSystem.yaml

```diff
@@ -36,48 +36,4 @@ spec:
-
-    violation[result] {
-      isExcludedNamespace == false
-      not exclude_label_value == controller_input.metadata.labels[exclude_label_key]
-      some i
-      containers := controller_spec.containers[i]
-      root_fs := containers.securityContext.readOnlyRootFilesystem
-      not root_fs == read_only
-      result = {
-        "issue detected": true,
-        "msg": sprintf("readOnlyRootFilesystem should equal '%v'; detected '%v'", [read_only, root_fs]),
-        "recommended_value": read_only,
-        "violating_key": sprintf("spec.template.spec.containers[%v].securityContext.readOnlyRootFilesystem", [i])
-      }
-    }
-
-    # Controller input
-    controller_input = input.review.object
-
-    # controller_container acts as an iterator to get containers from the template
-    controller_spec = controller_input.spec.template.spec {
-      contains_kind(controller_input.kind, {"StatefulSet" , "DaemonSet", "Deployment", "Job"})
-    } else = controller_input.spec {
-      controller_input.kind == "Pod"
-    } else = controller_input.spec.jobTemplate.spec.template.spec {
-      controller_input.kind == "CronJob"
-    }
-
-    contains_kind(kind, kinds) {
-      kinds[_] = kind
-    }
-
-    isExcludedNamespace = true {
-      controller_input.metadata.namespace
-      controller_input.metadata.namespace in exclude_namespaces
-    } else = false
+  code: "package weave.advisor.podSecurity.enforce_ro_fs\n\nimport future.keywords.in\n\nread_only = input.parameters.read_only\nexclude_namespaces := input.parameters.exclude_namespaces\nexclude_label_key := input.parameters.exclude_label_key\nexclude_label_value := input.parameters.exclude_label_value\n\nviolation[result] {\n  isExcludedNamespace == false\n  not exclude_label_value == controller_input.metadata.labels[exclude_label_key]\n  some i\n  containers := controller_spec.containers[i]\n  root_fs := containers.securityContext.readOnlyRootFilesystem\n  not root_fs == read_only\n  result = {\n    \"issue detected\": true,\n    \"msg\": sprintf(\"readOnlyRootFilesystem should equal '%v'; detected '%v'\", [read_only, root_fs]),\n    \"recommended_value\": read_only,\n    \"violating_key\": sprintf(\"spec.template.spec.containers[%v].securityContext.readOnlyRootFilesystem\", [i]) \n  }\n}\n\n# Controller input\ncontroller_input = input.review.object\n\n# controller_container acts as an iterator to get containers from the template\ncontroller_spec = controller_input.spec.template.spec {\n  contains_kind(controller_input.kind, {\"StatefulSet\" , \"DaemonSet\", \"Deployment\", \"Job\"})\n} else = controller_input.spec {\n  controller_input.kind == \"Pod\"\n} else = controller_input.spec.jobTemplate.spec.template.spec {\n  controller_input.kind == \"CronJob\"\n}\n\ncontains_kind(kind, kinds) {\n  kinds[_] = kind\n}\n\nisExcludedNamespace = true {\n\tcontroller_input.metadata.namespace\n\tcontroller_input.metadata.namespace in exclude_namespaces\n} else = false\n"
```

Comment:
I think something went wrong with the format. Prefer to keep the format consistent with the rest of policies

Reply:
It's taken from the policy library where the CI workflow handles this part. Let me see what can be done.